bee-bundle/src/incoming_bundle_builder.rs

```diff
@@ -122,21 +124,30 @@ where
         // TODO - check trunk of the last transaction and branch is tail, the same tail
 
         for (index, transaction) in self.transactions.0.iter().enumerate() {
-            if index != transaction.index().0 {
-                Err(IncomingBundleBuilderError::InvalidIndex(transaction.index().0))?;
+            if index != *transaction.index().into_inner() {
+                Err(IncomingBundleBuilderError::InvalidIndex(
+                    *transaction.index().into_inner(),
+                ))?;
             }
 
-            if last_index != transaction.last_index().0 {
-                Err(IncomingBundleBuilderError::InvalidLastIndex(transaction.last_index().0))?;
+            if last_index != *transaction.last_index().into_inner() {
+                Err(IncomingBundleBuilderError::InvalidLastIndex(
+                    *transaction.last_index().into_inner(),
+                ))?;
             }
 
-            sum += transaction.value.0;
+            sum += *transaction.value.into_inner();
             if sum.abs() > IOTA_SUPPLY {
                 Err(IncomingBundleBuilderError::InvalidValue(sum))?;
             }
 
-            if transaction.value.0 != 0
-                && transaction.address().0.get(ADDRESS.trit_offset.length - 1).unwrap() != Btrit::Zero
+            if *transaction.value.into_inner() != 0
+                && transaction
+                    .address()
+                    .into_inner()
+                    .get(ADDRESS.trit_offset.length - 1)
+                    .unwrap()
```

Comment:
Is `unwrap`ing here always safe? If it is, we should provide a small comment noting that.